Running a federation with GaNDLF Documentation

psfoley opened the pull request:
This PR:
- Adds a new documentation page that summarizes how to bring a GaNDLF experiment into OpenFL
- Adds a new flag to `fx plan initialize`  to import an existing GaNDLF config
    - `fx plan initialize --gandlf_config {GANDLF_CONFIG}.yaml`
- Refactors the training and validation CSV paths from the plan into the `data.yaml` file for consistency with other OpenFL usage

Breaking changes:
- The FeTS Challenge repo may need to be modified to support new CSV loading scheme (or pinned to earlier OpenFL version)

Todo:
- [ ] Differentiate simulation vs. real federation sections
- [x] Fix CI
- [x] Get signoff from @sarthakpati and Ujjwal Baid

psfoley committed:
Initial commit of Running the federation with GaNDLF Documentation

Signed-off-by: Patrick Foley <psfoley@gmail.com>

sarthakpati commented on docs/running_the_federation_with_gandlf.rst:
```diff
@@ -0,0 +1,508 @@
+
+
+`Aggregator-Based Workflow`_
+    Define an experiment and distribute it manually. All participants can verify model code and [FL plan](https://openfl.readthedocs.io/en/latest/running_the_federation.html#federated-learning-plan-fl-plan-settings) prior to execution. The federation is terminated when the experiment is finished
```
```suggestion
    Define an experiment and distribute it manually. All participants can verify model code and [FL plan](https://openfl.readthedocs.io/en/latest/running_the_federation.html#federated-learning-plan-fl-plan-settings) prior to executing the code/model. The federation is terminated when the experiment is finished, and appropriate statistics are generated.
```

sarthakpati commented on docs/running_the_federation_with_gandlf.rst:
```diff
@@ -0,0 +1,508 @@
+
+This workflow uses short-lived components in a federation, which is terminated when the experiment is finished. The components are as follows:
+
+- The *Collaborator* uses a local dataset to train a global model and the *Aggregator* receives model updates from *Collaborators* and aggregates them to create the new global model.
```
```suggestion
- The *Collaborator* uses a local dataset to train a global model and sends the model updates to the *Aggregator*, which aggregates them to create the new global model.
```

sarthakpati commented on docs/running_the_federation_with_gandlf.rst:
```diff
@@ -0,0 +1,508 @@
+This workflow uses short-lived components in a federation, which is terminated when the experiment is finished. The components are as follows:
+
+- The *Collaborator* uses a local dataset to train a global model and the *Aggregator* receives model updates from *Collaborators* and aggregates them to create the new global model.
+- The *Aggregator* is framework-agnostic, while the *Collaborator* can use any deep learning frameworks, such as `TensorFlow <https://www.tensorflow.org/>`_\* \  or `PyTorch <https://pytorch.org/>`_\*\, or `GaNDLF <https://github.com/mlcommons/GaNDLF>`
```
```suggestion
- The *Aggregator* is framework-agnostic, while the *Collaborator* can use any deep learning frameworks, such as `TensorFlow <https://www.tensorflow.org/>`_\* \  or `PyTorch <https://pytorch.org/>`_\*\. `GaNDLF <https://github.com/mlcommons/GaNDLF>` provides a straightforward way to define complete model training pipelines for healthcare data, and is directly compatible with OpenFL.
```

sarthakpati commented on docs/running_the_federation_with_gandlf.rst:
```diff
@@ -0,0 +1,508 @@
+
+******************************
+Run the Federation with GaNDLF
+******************************
```
```suggestion
****************************************************
Run the Federation with a model defined using GaNDLF
****************************************************
```

sarthakpati commented on docs/running_the_federation_with_gandlf.rst:
```diff
@@ -0,0 +1,508 @@
+Run the Federation with GaNDLF
+******************************
+
+This guide will show you how to take an existing GaNDLF experiment to a federated environment. 
```
```suggestion
This guide will show you how to take an existing model using the [Generally Nuanced Deep Learning Framework (GaNDLF)](https://github.com/mlcommons/GaNDLF) experiment to a federated environment. 
```

sarthakpati commented on docs/running_the_federation_with_gandlf.rst:
```diff
@@ -0,0 +1,508 @@
+- The *Collaborator* uses a local dataset to train a global model and the *Aggregator* receives model updates from *Collaborators* and aggregates them to create the new global model.
+- The *Aggregator* is framework-agnostic, while the *Collaborator* can use any deep learning frameworks, such as `TensorFlow <https://www.tensorflow.org/>`_\* \  or `PyTorch <https://pytorch.org/>`_\*\, or `GaNDLF <https://github.com/mlcommons/GaNDLF>`
+
+This guide will demonstrate how to take an existing GaNDLF configuration file (i.e. for segmentation), embed this within the Federated Learning plan (FL plan) along with the Python\*\  code that defines the model and the data loader. The FL plan is a `YAML <https://en.wikipedia.org/wiki/YAML>`_ file that defines the collaborators, aggregator, connections, models, data, and any other parameters th ...
```
```suggestion
This guide will demonstrate how to take an existing [GaNDLF model configuration](https://mlcommons.github.io/GaNDLF/getting_started/) (e.g., for segmentation), embed this within the Federated Learning plan (FL plan) along with the Python\*\  code that defines the model and the data loader. The FL plan is a `YAML <https://en.wikipedia.org/wiki/YAML>`_ file that defines the collaborators, aggregator, connections, models, data, and any other parameters that describe the training.
```

sarthakpati commented on docs/running_the_federation_with_gandlf.rst:
```diff
@@ -0,0 +1,508 @@
+        git checkout 0.0.14
+        pip install torch==1.8.2+cpu torchvision==0.9.2+cpu torchaudio===0.8.2 -f https://download.pytorch.org/whl/lts/1.8/torch_lts.html
+        pip install onnx==1.12.0
+        pip install -e .
```
Removing explicit instructions because these are already old (for 0.0.14), and I think it would make sense to have GaNDLF-specific installation instructions in a single place.
```suggestion
2.     [Install GaNDLF from sources](https://mlcommons.github.io/GaNDLF/setup/#install-from-sources) (if not already).
```

sarthakpati commented on docs/running_the_federation_with_gandlf.rst:
```diff
@@ -0,0 +1,508 @@
+        pip install onnx==1.12.0
+        pip install -e .
+
+3.     Create GaNDLF's Data CSVs for 3D Segmentation
```
```suggestion
3.     Create GaNDLF's Data CSVs. The example below is for 3D Segmentation using the unit test data:
```

sarthakpati commented on docs/running_the_federation_with_gandlf.rst:
```diff
@@ -0,0 +1,508 @@
+        git checkout 0.0.14
+        pip install torch==1.8.2+cpu torchvision==0.9.2+cpu torchaudio===0.8.2 -f https://download.pytorch.org/whl/lts/1.8/torch_lts.html
+        pip install onnx==1.12.0
+        pip install -e .
```
(Same as above) Removing explicit instructions because these are already old (for 0.0.14), and I think it would make sense to have GaNDLF-specific installation instructions in a single place.
```suggestion
2.     [Install GaNDLF from sources](https://mlcommons.github.io/GaNDLF/setup/#install-from-sources) (if not already).
```

sarthakpati commented on docs/running_the_federation_with_gandlf.rst:
```diff
@@ -0,0 +1,508 @@
+Post Experiment
+^^^^^^^^^^^^^^^
+
+Experiment owners may access the final model in its native format.
```
```suggestion
Experiment owners may access the final model in its native format. Once the model has been converted to its native format, inference can be done using [GaNDLF's inference API](https://mlcommons.github.io/GaNDLF/usage/#running-gandlf-traininginference).
```

sarthakpati reviewed (changes requested):
This looks great, thank you so much, @psfoley!

I have added a few comments that (hopefully) make the documentation clearer, and ensure less maintenance for the OpenFL team. Additionally, I would also recommend that we update the [fets challenge workflow](https://github.com/securefederatedai/openfl/blob/develop/.github/workflows/fets-challenge.yml) to point to [MLCommons/GaNDLF instead of CBICA/GaNDLF](https://github.com/securefederatedai/openfl/blob/develop/.github/workflows/fets-challenge.yml#L31), and to use the latest tag [instead of 0.0.14](https://github.com/securefederatedai/openfl/blob/develop/.github/workflows/fets-challenge.yml#L32). [GaNDLF already does this for OpenFL](https://github.com/mlcommons/GaNDLF/blob/master/.github/workflows/openfl-test.yml#L73).

psfoley committed:
Update docs/running_the_federation_with_gandlf.rst

Co-authored-by: Sarthak Pati <sarthak.pati@pennmedicine.upenn.edu>
Signed-off-by: Patrick Foley <psfoley@gmail.com>

psfoley committed:
Update docs/running_the_federation_with_gandlf.rst

Co-authored-by: Sarthak Pati <sarthak.pati@pennmedicine.upenn.edu>
Signed-off-by: Patrick Foley <psfoley@gmail.com>

psfoley committed:
Update docs/running_the_federation_with_gandlf.rst

Co-authored-by: Sarthak Pati <sarthak.pati@pennmedicine.upenn.edu>
Signed-off-by: Patrick Foley <psfoley@gmail.com>

psfoley committed:
Update docs/running_the_federation_with_gandlf.rst

Co-authored-by: Sarthak Pati <sarthak.pati@pennmedicine.upenn.edu>
Signed-off-by: Patrick Foley <psfoley@gmail.com>

psfoley committed:
Update docs/running_the_federation_with_gandlf.rst

Co-authored-by: Sarthak Pati <sarthak.pati@pennmedicine.upenn.edu>
Signed-off-by: Patrick Foley <psfoley@gmail.com>

psfoley committed:
Update docs/running_the_federation_with_gandlf.rst

Co-authored-by: Sarthak Pati <sarthak.pati@pennmedicine.upenn.edu>
Signed-off-by: Patrick Foley <psfoley@gmail.com>

psfoley committed:
Update docs/running_the_federation_with_gandlf.rst

Co-authored-by: Sarthak Pati <sarthak.pati@pennmedicine.upenn.edu>
Signed-off-by: Patrick Foley <psfoley@gmail.com>

psfoley committed:
Update docs/running_the_federation_with_gandlf.rst

Co-authored-by: Sarthak Pati <sarthak.pati@pennmedicine.upenn.edu>
Signed-off-by: Patrick Foley <psfoley@gmail.com>

psfoley committed:
Update docs/running_the_federation_with_gandlf.rst

Co-authored-by: Sarthak Pati <sarthak.pati@pennmedicine.upenn.edu>
Signed-off-by: Patrick Foley <psfoley@gmail.com>

psfoley committed:
Update docs/running_the_federation_with_gandlf.rst

Co-authored-by: Sarthak Pati <sarthak.pati@pennmedicine.upenn.edu>
Signed-off-by: Patrick Foley <psfoley@gmail.com>

SprashAI committed:
Update README.md

Removed references to Intel's ownship, given it's now owned by the LF AI and Data.

Signed-off-by: Patrick Foley <psfoley@gmail.com>

akantak committed:
Fix Flake8 C419 for Ubuntu CI (#800)

C419 Unnecessary list comprehension passed to any()/all() prevents short-circuiting - rewrite as a generator

Signed-off-by: Aleksander Kantak <aleksander.kantak@intel.com>
Signed-off-by: Patrick Foley <psfoley@gmail.com>

psfoley committed:
Update README.md

Signed-off-by: Patrick Foley <psfoley@gmail.com>

akantak committed:
Fix warnings and issues in docs (#825)

* Fix warnings and issues in docs

Signed-off-by: Aleksander Kantak <aleksander.kantak@intel.com>

* fixup! Fix warnings and issues in docs

Signed-off-by: Aleksander Kantak <aleksander.kantak@intel.com>

---------

Signed-off-by: Aleksander Kantak <aleksander.kantak@intel.com>
Signed-off-by: Patrick Foley <psfoley@gmail.com>

psfoley committed:
Add Logo (#827)

* Add Logo

* Update README.md

Signed-off-by: Patrick Foley <psfoley@gmail.com>

wangleflex committed:
Change OpenFL documentation font to improve accessibility (#809)

This replaces the font of OpenFL documents with Intel One Mono font for low vision developers.

    Known issues:
    1. The text font within the images has not been changed.
    2. Some icons that do not exist in the new font cannot be displayed properly.

    Fixes securefederatedai#799

    Co-authored-by: Wang, Le <le3.wang@intel.com>
    Signed-off-by: He, Dan H <dan.h.he@intel.com>
    Signed-off-by: Jiang, Jiaqiu <jiaqiu.jiang@intel.com>
    Signed-off-by: Li, Qingqing <qingqing.li@intel.com>
    Signed-off-by: Wang, Le <le3.wang@intel.com>
    Signed-off-by: Wu, Caili <caili.wu@intel.com>

Signed-off-by: He, Dan H <dan.h.he@intel.com>
Co-authored-by: He, Dan H <dan.h.he@intel.com>
Signed-off-by: Patrick Foley <psfoley@gmail.com>

fangxiaoran committed:
Update unit tests to improve code coverage (#821)

* Update ci config

Signed-off-by: Fang, Xiaoran <xiaoran.fang@intel.com>

* Add unit test for following files

- openfl/federated/plan/plan.py
- openfl/interface/aggregation_functions/core/adaptive_aggregation.py

Signed-off-by: Fang, Xiaoran <xiaoran.fang@intel.com>

* Add some test cases for databases module

Signed-off-by: Fang, Xiaoran <xiaoran.fang@intel.com>

* Fix bugs for databases module unittest

Signed-off-by: Fang, Xiaoran <xiaoran.fang@intel.com>

* Update unit tests for component module

Signed-off-by: Fang, Xiaoran <xiaoran.fang@intel.com>

* Restore workflow config and update some comments

Signed-off-by: Fang, Xiaoran <xiaoran.fang@intel.com>

* Enable save_ test case.
Add yaml under test dir for unit test usage.

Signed-off-by: Fang, Xiaoran <xiaoran.fang@intel.com>

* Remove plan to new dir.

Signed-off-by: Fang, Xiaoran <xiaoran.fang@intel.com>

* Remove plan to new dir.

Signed-off-by: Fang, Xiaoran <xiaoran.fang@intel.com>

* Add aggregator start test cases.

Signed-off-by: Fang, Xiaoran <xiaoran.fang@intel.com>

* Add 2 aggregator test cases.

Signed-off-by: Fang, Xiaoran <xiaoran.fang@intel.com>

* Add 1 aggregator test case.

Signed-off-by: Fang, Xiaoran <xiaoran.fang@intel.com>

* Format code.

Signed-off-by: Fang, Xiaoran <xiaoran.fang@intel.com>

* Refactor code.

Signed-off-by: Fang, Xiaoran <xiaoran.fang@intel.com>

* Add collaborator start test cases.

Signed-off-by: Fang, Xiaoran <xiaoran.fang@intel.com>

* Add 1 collaborator test case.

Signed-off-by: Fang, Xiaoran <xiaoran.fang@intel.com>

* Format with flake8

Signed-off-by: Fang, Xiaoran <xiaoran.fang@intel.com>

* Remove TODO comments

Signed-off-by: Fang, Xiaoran <xiaoran.fang@intel.com>

---------

Signed-off-by: Fang, Xiaoran <xiaoran.fang@intel.com>
Co-authored-by: Wang, Wenjie <wenjie2.wang@intel.com>
Co-authored-by: Lei5 Chen <lei5.chen@intel.com>
Signed-off-by: Patrick Foley <psfoley@gmail.com>

danhe1 committed:
Add PyTorch linear regression example (#808)

This adds a new tutorial example on distributing a linear regression task over OpenFL cluster.

    The model is defined by Pytorch which is able to run over both cpu (by default) and gpu. The dataset is generated by make_regression from sklearn.datasets with pre-defined parameters.

    Fixes #797

    Co-authored-by: Jiang, Jiaqiu <jiaqiu.jiang@intel.com>
    Signed-off-by: He, Dan H <dan.h.he@intel.com>
    Signed-off-by: Jiang, Jiaqiu <jiaqiu.jiang@intel.com>
    Signed-off-by: Li, Qingqing <qingqing.li@intel.com>
    Signed-off-by: Wang, Le <le3.wang@intel.com>
    Signed-off-by: Wu, Caili <caili.wu@intel.com>

Signed-off-by: He, Dan H <dan.h.he@intel.com>
Signed-off-by: Patrick Foley <psfoley@gmail.com>

bjklemme-intel committed:
This prints out the hash of the CSR to disk for both the aggregator and (#813)

* This prints out the hash of the CSR to disk for both the aggregator and
collaborator. The user then compares and approves this hash with the
hash printed out of the file to validate the CSR. In addition, a warning
message is pritned if certify is run in silent mode.

Fixes securefederatedai#692

Signed-off-by: Grant Baker <grant.baker@intel.com>

* Refactor read_csr function to use get_csr_hash

Signed-off-by: Grant Baker <grant.baker@intel.com>

* Ask to check hashes before prompt

---------

Signed-off-by: Grant Baker <grant.baker@intel.com>
Co-authored-by: Grant Baker <grant.baker@intel.com>

Signed-off-by: Patrick Foley <psfoley@gmail.com>

danhe1 committed:
Improve workspace requirements import (#810)

Remove the dump_requirement_file operation in export_ method.

            Fixes securefederatedai#767

            Co-authored-by: Li, Qingqing <qingqing.li@intel.com>
            Co-authored-by: Wu, Caili <caili.wu@intel.com>
            Signed-off-by: He, Dan H <dan.h.he@intel.com>
            Signed-off-by: Jiang, Jiaqiu <jiaqiu.jiang@intel.com>
            Signed-off-by: Li, Qingqing <qingqing.li@intel.com>
            Signed-off-by: Wang, Le <le3.wang@intel.com>
            Signed-off-by: Wu, Caili <caili.wu@intel.com>

Signed-off-by: He, Dan H <dan.h.he@intel.com>
Signed-off-by: Patrick Foley <psfoley@gmail.com>

bjklemme-intel committed:
Issue 506 Added Example using FedProx (#818)

* created new ineractive_api dir to hold pytorch fedprox mnist example

corrected files

changed to FedProxOptimizer and ran set_old_weights for new FedProx Pytorch example

renamed FedProx notebook

used mode.parrameters() to get pytorch model weights

got weights using state_dict

changed old wieghts to list (for serialization) and fixed README

input wieghts before zero_grad

[Enhancement: 506] Add an example that uses the FedProx optimizer in the interative_api

This duplicates the MedNIST_2D example in the interative api but changes it to use the FedProx optimizer.

Fixes: #506

Signed-off-by: Klemme, Beverly <beverly.klemme@intel.com>
Signed-off-by: Baker, Grant <grant.baker@intel.com>
Signed-off-by: ELizabeth Simon, Neethu <neethu.elizabeth.simon@intel.com>
Signed-off-by: Jillela, Emmanuel <emmanuel.jillela@intel.com>
Signed-off-by: Beverly Klemme <beverly.klemme@intel.com>

* [Enhancement: 506] Add an example that uses the FedProx optimizer in the interative_api

This duplicates the MedNIST_2D example in the interative api but changes it to use the FedProx optimizer.

Fixes: securefederatedai#506

Signed-off-by: Klemme, Beverly <beverly.klemme@intel.com>
Signed-off-by: Baker, Grant <grant.baker@intel.com>
Signed-off-by: Elizabeth Simon, Neethu <neethu.elizabeth.simon@intel.com>
Signed-off-by: Jillela, Emmanuel <emmanuel.jillela@intel.com>
Signed-off-by: Beverly Klemme <beverly.klemme@intel.com>

* Update README.md

Signed-off-by: Beverly Klemme <beverly.klemme@intel.com>

* addressed comments by psfoley: corrected words in the jupyter notebook metadata and added a link to the FedProx paper in the README.

Signed-off-by: Beverly Klemme <beverly.klemme@intel.com>

---------

Signed-off-by: Klemme, Beverly <beverly.klemme@intel.com>
Signed-off-by: Baker, Grant <grant.baker@intel.com>
Signed-off-by: ELizabeth Simon, Neethu <neethu.elizabeth.simon@intel.com>
Signed-off-by: Jillela, Emmanuel <emmanuel.jillela@intel.com>
Signed-off-by: Beverly Klemme <beverly.klemme@intel.com>
Signed-off-by: Elizabeth Simon, Neethu <neethu.elizabeth.simon@intel.com>
Signed-off-by: Patrick Foley <psfoley@gmail.com>

bjklemme-intel committed:
[Bug: 768] FX CLI: Separate create, cert gen commands (#807)

This change separates existing command
"fx collaborator.py generate-cert-request" command into two
commands.
"fx collaborator create -n {NAME} -d {DATA_PATH: optional}".
"fx collaborator generate-cert-request -n {NAME}".

Fixes #768

Signed-off-by: Emmanuel Jillela <emmanuel.jillela@intel.com>
Co-authored-by: Emmanuel Jillela <emmanuel.jillela@intel.com>
Signed-off-by: Patrick Foley <psfoley@gmail.com>

bjklemme-intel committed:
Add new tutorial example to OpenFL interactive API (#812)

* Add new tutorial example to OpenFL interactive API

This adds a new tutorial example on distributing a linear regression task over OpenFL cluster

The model is defined by scikit-learn which is able to run over both cpu (by default) and gpu. The dataset is 1-dimensional noisy data of sinusoid with pre-defined parameters.

Fixes #798

Co-authored-by: Beverly Klemme <beverly.klemme@intel.com>
Co-authored-by: Grant Baker <grant.baker@intel.com>

Signed-off-by: Yi CAO <yi2.cao@intel.com>

* reduced requirements.txt in workspace

Signed-off-by: Beverly Klemme <beverly.klemme@intel.com>

---------

Signed-off-by: Yi CAO <yi2.cao@intel.com>
Signed-off-by: Beverly Klemme <beverly.klemme@intel.com>
Co-authored-by: Yi CAO <yi2.cao@intel.com>
Signed-off-by: Patrick Foley <psfoley@gmail.com>

dependabot[bot] committed:
build(deps): bump tensorflow in /openfl-workspace/tf_cnn_histology (#776)

Bumps [tensorflow](https://github.com/tensorflow/tensorflow) from 2.9.3 to 2.11.1.
- [Release notes](https://github.com/tensorflow/tensorflow/releases)
- [Changelog](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md)
- [Commits](https://github.com/tensorflow/tensorflow/compare/v2.9.3...v2.11.1)

---
updated-dependencies:
- dependency-name: tensorflow
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Signed-off-by: Patrick Foley <psfoley@gmail.com>

dependabot[bot] committed:
build(deps): bump tensorflow (#777)

Bumps [tensorflow](https://github.com/tensorflow/tensorflow) from 2.9.3 to 2.11.1.
- [Release notes](https://github.com/tensorflow/tensorflow/releases)
- [Changelog](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md)
- [Commits](https://github.com/tensorflow/tensorflow/compare/v2.9.3...v2.11.1)

---
updated-dependencies:
- dependency-name: tensorflow
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Signed-off-by: Patrick Foley <psfoley@gmail.com>

psfoley committed:
Update GaNDLF repo location and test CI with master branch

Signed-off-by: Patrick Foley <psfoley@gmail.com>

psfoley committed:
Update GaNDLF repo location and test CI with master branch

Signed-off-by: Patrick Foley <psfoley@gmail.com>

psfoley committed:
Update GaNDLF repo location and test CI with master branch

Signed-off-by: Patrick Foley <psfoley@gmail.com>

psfoley committed:
Fix documentation links. Change path names and templates for CI

Signed-off-by: Patrick Foley <psfoley@gmail.com>

psfoley committed:
Fix paths

Signed-off-by: Patrick Foley <psfoley@gmail.com>

psfoley committed:
Fix paths

Signed-off-by: Patrick Foley <psfoley@gmail.com>

psfoley committed:
Fix breaking tests

Signed-off-by: Patrick Foley <psfoley@gmail.com>

psfoley committed:
Add compatible onnx version to requirements.txt file

Signed-off-by: Patrick Foley <psfoley@gmail.com>

psfoley committed:
Fix wrong csv file name

Signed-off-by: Patrick Foley <psfoley@gmail.com>

psfoley committed:
Fix wrong csv file name

Signed-off-by: Patrick Foley <psfoley@gmail.com>

psfoley committed:
Fix wrong names in workflow file

Signed-off-by: Patrick Foley <psfoley@gmail.com>

psfoley committed:
Fix wrong data path

Signed-off-by: Patrick Foley <psfoley@gmail.com>

psfoley committed:
Fix lint in test_gandlf.py

Signed-off-by: Patrick Foley <psfoley@gmail.com>

psfoley committed:
Fix lint errors

Signed-off-by: Patrick Foley <psfoley@gmail.com>
